src/Microsoft.OpenApi/Services/OpenApiReferenceResolver.cs

```diff
@@ -200,17 +200,17 @@ public override void Visit(ref JsonSchema schema)
             var description = schema.GetDescription();
             var summary = schema.GetSummary();
 
-            if (reference != null)
+            if (schema.Keywords.Count.Equals(1) && reference != null)
             {
                 schema = ResolveJsonSchemaReference(reference, description, summary);
             }
-            
+
             var builder = new JsonSchemaBuilder();
-            foreach (var keyword in schema.Keywords)
+            foreach (var keyword in schema?.Keywords)
```

Comment:
```suggestion
            if (schema?.Keywords is {} keywords)
            {
                 foreach (var keyword in keywords)
```

This will cause a null reference exception with the current form.

(and add missing closing curly + fix indent)